tool/tctl/common/auth_command.go

```diff
@@ -236,6 +243,48 @@ func (a *AuthCommand) GenerateAndSignKeys(ctx context.Context, clusterAPI auth.C
 	}
 }
 
+func (a *AuthCommand) generateWindowsCert(ctx context.Context, clusterAPI auth.ClientI) error {
+	var missingFlags []string
+	if len(a.windowsUser) == 0 {
+		missingFlags = append(missingFlags, "--windows-user")
+	}
+	if len(a.windowsDomain) == 0 {
+		missingFlags = append(missingFlags, "--windows-domain")
+	}
+	if len(missingFlags) > 0 {
+		return trace.BadParameter("the following flags are missing: %v",
+			strings.Join(missingFlags, ", "))
+	}
+
+	cn, err := clusterAPI.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	certDER, _, err := windows.GenerateCredentials(ctx, a.windowsUser, a.windowsDomain, a.genTTL,
+		cn.GetClusterName(), windows.LDAPConfig{Domain: a.windowsDomain}, clusterAPI)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = identityfile.Write(identityfile.WriteConfig{
+		OutputPath: a.output,
+		Key: &client.Key{
+			// the godocs say the map key is the desktop server name,
+			// but in this case we're just generating a cert that's not
+			// specific to a particular desktop
+			WindowsDesktopCerts: map[string][]byte{a.windowsUser: certDER},
+		},
+		Format:               a.outputFormat,
```

Comment:
Very nitpicky, but does it make sense to use the constant here? I think this will only ever be one value given the switch above?

```suggestion
		Format:               identityfile.FormatWindows,
```